warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -459,15 +512,19 @@ func (as *AzureSynapse) ProcessColumnValue(
 	case model.BooleanDataType:
 		return strconv.ParseBool(value)
 	case model.StringDataType:
-		if len(value) > stringLengthLimit {
-			value = value[:stringLengthLimit]
+		if varcharLength == varcharMaxLength {
```

Comment:
when would we get `varcharMaxLength` as in `-1`? Is it how it returns the value from getVarcharLengths?

Reply:
We can increase the size to 8000 using VARCHAR(8000) post then we can only set it as VARCHAR(MAX) which has a storage space of 2 GB. Querying the `INFORMATION_SCHEMA.COLUMNS` returns -1 in that case.

https://learn.microsoft.com/en-us/sql/t-sql/data-types/char-and-varchar-transact-sql?view=sql-server-ver16#varchar---n--max--

Reply:
We can add it as a comment in the code